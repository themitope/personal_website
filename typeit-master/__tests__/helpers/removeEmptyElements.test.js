import removeEmptyElements from "../../src/helpers/removeEmptyElements";

test("Empty HTML nodes should be removed from DOM.", () => {
  setHTML`<span id="scope">a<span></span>bc</span>`;
  removeEmptyElements(document.getElementById("scope"));
  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Nested empty HTML nodes should be removed from DOM.", () => {
  setHTML`<span id="scope">ab<span><a></a></span>c</span>`;

  removeEmptyElements(document.getElementById("scope"));
  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Multiple empty HTML nodes should be removed from DOM.", () => {
  setHTML`<span id="scope">a<span></span><span></span>b<em></em>c</span>`;

  removeEmptyElements(document.getElementById("scope"));
  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Multiple nested empty HTML nodes should be removed from DOM.", () => {
  setHTML`<span id="scope">a<span><a><em></em></a></span>b<span><a><em></em></a></span>c</span>`;

  removeEmptyElements(document.getElementById("scope"));
  expect(document.body.innerHTML).toMatchSnapshot();
});

test("Break tags should not be removed.", () => {
  setHTML`<span id="scope">First line.<br>Second line.<br><br/>Third line.</span>`;
  removeEmptyElements(document.getElementById("scope"));
  expect(document.body.innerHTML).toBe(
    '<span id="scope">First line.<br>Second line.<br><br>Third line.</span>'
  );
});
