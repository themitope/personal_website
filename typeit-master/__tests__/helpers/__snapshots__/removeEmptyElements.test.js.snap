// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty HTML nodes should be removed from DOM. 1`] = `"<span id=\\"scope\\">abc</span>"`;

exports[`Multiple empty HTML nodes should be removed from DOM. 1`] = `"<span id=\\"scope\\">abc</span>"`;

exports[`Multiple nested empty HTML nodes should be removed from DOM. 1`] = `"<span id=\\"scope\\">abc</span>"`;

exports[`Nested empty HTML nodes should be removed from DOM. 1`] = `"<span id=\\"scope\\">abc</span>"`;
